nibabies/workflows/anatomical/base.py

```diff
@@ -200,53 +210,48 @@ def init_infant_anat_wf(
         cifti_output=cifti_output,
     )
 
-    # Multiple T1w files -> generate average reference
-    t1w_template_wf = init_anat_average_wf(
+    # Multiple anatomical files -> generate average reference
+    t1w_template_wf = init_anat_template_wf(
+        contrast="T1w",
+        num_files=num_t1w,
         longitudinal=longitudinal,
         omp_nthreads=omp_nthreads,
-        num_maps=num_t1w,
+        sloppy=sloppy,
+        precomputed_mask=bool(precomp_mask),
+        precomputed_aseg=bool(precomp_aseg),
         name="t1w_template_wf",
     )
 
-    t2w_template_wf = init_anat_average_wf(
+    t2w_template_wf = init_anat_template_wf(
+        contrast="T2w",
+        num_files=num_t2w,
         longitudinal=longitudinal,
         omp_nthreads=omp_nthreads,
-        num_maps=num_t2w,
+        sloppy=sloppy,
         name="t2w_template_wf",
     )
 
-    # INU + Brain Extraction
+    # Clean up each anatomical template
+    # Denoise, INU, + Clipping
+    t1w_preproc_wf = init_anat_preproc_wf(name="t1w_preproc_wf")
+    t2w_preproc_wf = init_anat_preproc_wf(name="t2w_preproc_wf")
```

Comment:
Are we making strong assumptions that `num_t1w` and `num_t2w` >0? Even if it's true for now, I would consider guarding these.